src/components/AppArkApiTable/AppArkApiTable.tsx

```diff
@@ -54,23 +69,52 @@ export function PaginatedSortableTable<T>(props: PaginatedSortableTableProps<T>)
     externalFilters,
     externalFiltersState,
     isSortable = true,
+    setRowSelection,
+    selectedRows,
+    pageSize: pageSizeProp,
+    getRowId,
+    defaultSorting,
+    skip = false,
+    extractPagination,
+    tableKey,
   } = props;
   const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
     pageIndex: 1,
-    pageSize: 10,
+    pageSize: pageSizeProp ?? 10,
   });
-
-  const [sortingState, setSorting] = useState<SortingState>([{ id: "", desc: false }]);
-
+  const reduxTableState = useAppSelector(state => getTableState(state, tableKey ?? ""));
+  const filtersAreEquals = JSON.stringify(reduxTableState?.filters) === JSON.stringify(externalFiltersState);
```

Comment:
this is asking for trouble as there is no check that the filters nor the externalFiltersState are serializable as the interface supports extensions.

you should slice only the fields expected by interface before serialization.